crates/net/network/src/manager.rs

```diff
@@ -951,6 +956,9 @@ pub enum NetworkEvent {
         capabilities: Arc<Capabilities>,
         /// A request channel to the session task.
         messages: PeerRequestSender,
+        /// Counting semaphore for inflight requests read by
+        /// [`crate::transactions::TransactionsManager`].
+        inflight_requests_semaphore_rx: watch::Receiver<usize>,
```

Comment:
I think we should move this var into the `PeerRequestSender` struct.

https://github.com/paradigmxyz/reth/blob/df4774c709c906976842425f294f5e3f8fe876df/crates/net/network/src/message.rs#L310-L317

Reply:
Totally